elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCleanerThread.java

```diff
@@ -0,0 +1,140 @@
+                        .filterExpression(filter)
+                        .build();
+
+                Iterable<Object> loadedObjects = tx.loadObjects(asyncQueryCollection, scope);
+                return loadedObjects;
+            } catch (Exception e) {
+                log.error("Exception: {}", e.getMessage());
+            }
+            return null;
+        });
+        Iterator<Object> itr = loaded.iterator();
+        while(itr.hasNext()) {
+            AsyncQuery query = (AsyncQuery) itr.next();
+
+            log.info("Found a query to DELETE");
+            asyncDbUtil.deleteAsyncQueryAndResult(query.getId());
+        }
+    }
+    
+    /**
+     * This method updates the status of long running async query which
+     * were not interrupted due to host crash/app shutdown to TIMEDOUT.
+     * */
+	@SuppressWarnings("unchecked")
+    private void timeoutAsyncQuery() {
+        AsyncDbUtil asyncDbUtil = AsyncDbUtil.getInstance(elide);
+        Iterable<Object> loaded = (Iterable<Object>) asyncDbUtil.executeInTransaction(elide.getDataStore(), (tx, scope) -> {
+            try {
+                EntityDictionary dictionary = elide.getElideSettings().getDictionary();
+                RSQLFilterDialect filterParser = new RSQLFilterDialect(dictionary);
+                FilterExpression filter = filterParser.parseFilterExpression("status=in=(" + QueryStatus.PROCESSING.toString() + ","
+                        + QueryStatus.QUEUED.toString() + ")", AsyncQuery.class, false);
+
+                EntityProjection asyncQueryCollection = EntityProjection.builder()
+                        .type(AsyncQuery.class)
+                        .filterExpression(filter)
+                        .build();
+
+                Iterable<Object> loadedObj = tx.loadObjects(asyncQueryCollection, scope);
+                return loadedObj;
+            } catch (Exception e) {
+                log.error("Exception: {}", e.getMessage());
+            }
+            return null;
+        });
+        Iterator<Object> itr = loaded.iterator();
+        long currentTime = new Date().getTime();
+        while(itr.hasNext()) {
+            AsyncQuery query = (AsyncQuery) itr.next();
+
+            if(isTimedOut(currentTime, query)) {
+                log.info("Updating Async Query Status to TIMEDOUT");
+                asyncDbUtil.updateAsyncQuery(query.getId(), (asyncQueryObj) -> {
+                    asyncQueryObj.setStatus(QueryStatus.TIMEDOUT);
+                    });
+            }
+        }
+    }
+
+    private boolean isTimedOut(long currentTime, AsyncQuery query) {
+        long differenceMillies = Math.abs(currentTime - query.getCreatedOn().getTime());
+        long differenceMinutes = TimeUnit.MINUTES.convert(differenceMillies, TimeUnit.MILLISECONDS);
+
+        // Check if its twice as long as max run time. It means the host/app crashed or restarted.
+        return (differenceMinutes > maxRunTimeMinutes * 2);
```

Comment:
Multiplying maxRunTimeMinutes by 2 should be done in a single place across all the code.